### ISSUE TITLE
Opening invalid sketch via file association/command line fails silently

### ISSUE THREAD

**Issue by Chose99** [arduino/arduino-ide#964]
### Describe the problem

An Arduino sketch is a folder. This folder must contain an `.ino` file matching the folder name:

```text
Foo/
└── Foo.ino
```

Users who have only worked with the single file sketches common in introductory tutorials and examples may have the impression that the file is the sketch. Sketch authors may distribute sketches as a file or inline code for convenience. Some distribution methods (e.g., GitHub's automatically generated ZIP files) may alter the name of the sketch folder. Any of these things may result in an invalid sketch due to the lack of alignment between folder and `.ino` file name.

For example:

```text
Foo/
└── Bar.ino
```

If I start the Arduino IDE by opening a sketch that does not contain a `.ino` file matching the folder name ([which is required for it to be considered valid](https://arduino.github.io/arduino-cli/latest/sketch-specification/#primary-sketch-file)), the IDE opens with the same behavior as it would without a sketch argument (last loaded sketches are opened).

### To reproduce

1. Create a sketch that does not meet the naming requirements.
   For example:
   ```text
   Foo/
   └── Bar.ino
   ```
1. Open the sketch directly, either from Windows "**File Explorer**" if you have `.ino` file association configured, or by running the IDE from the command line with the sketch path as an argument.
   ```text
   & ".\Arduino IDE" "C:\Users\per\Documents\Arduino\Foo\Bar.ino"
   ```

🐛 The IDE does not communicate the problem to the user. The user is more likely to think something is wrong with the IDE than with the sketch.
🐛 The IDE does not offer to move the sketch file to an appropriately named folder

### Expected behavior

The same behavior as when opening the file via **File > Open** in the Arduino IDE, as it works in Arduino IDE 1.x.

![image](https://user-images.githubusercontent.com/8572152/164736315-9b9c32e9-163e-4d97-aa66-42d149a5516a.png)

#### Original report

2.0.0-rc6-nightly-20220422

#### Last verified with

2.0.1-snapshot-8380c82

### Operating system

Windows

### Operating system version

10.0.19044

### Issue checklist

- [X] I searched for previous reports in [the issue tracker](https://github.com/arduino/arduino-ide/issues?q=)
- [X] I verified the problem still occurs when using the latest [nightly build](https://github.com/arduino/arduino-ide#nightly-builds)
- [X] My report contains all necessary details

**Comment by per1234** [comment]
>it does not work with valid sketches either.

Try it with the absolute sketch path @kittaakos. I get the same results from a relative path, but it works fine with the absolute sketch path.

**Comment by kittaakos** [comment]
Thank you for verifying it on Windows. It does not work with absolute path either on macOS. I tried these:

```
% ls -al ./my_workspace 
total 0
drwxr-xr-x   3 a.kitta  staff   96 Oct 19 10:41 .
drwx------@ 16 a.kitta  staff  512 Oct 19 11:56 ..
-rw-r--r--   1 a.kitta  staff    0 Oct 19 10:41 my_workspace.ino
```

```
% /Applications/Arduino\ IDE.app/Contents/MacOS/Arduino\ IDE /Users/a.kitta/Desktop/my_workspace 
```

```
% /Applications/Arduino\ IDE.app/Contents/MacOS/Arduino\ IDE /Users/a.kitta/Desktop/my_workspace/my_workspace.ino
```

Electron [requires different file-association implementations](https://github.com/arduino/arduino-ide/blob/87ebcbe77e38d3ce435758c6836d9ca07f86acd0/arduino-ide-extension/src/electron-main/theia/electron-main-application.ts#L141-L148) for macOS and other OSs. I bet that is causing the problem in IDE2, but I do not understand why relative workspace files are broken on Windows. It should work. See [here](https://github.com/eclipse-theia/theia/blob/ca417a31e402bd35717d3314bf6254049d1dae44/packages/workspace/src/node/default-workspace-server.ts#L50-L53).

This is just a guess, but I think a good one. The `.ino` file association implementation overrules Theia's default CLI contribution:
 - on macOS, and that's why it does not work. The frontend does not have the chance to pick up the workspace path from the CLI, but IDE2 forcefully tries to restore all previously opened sketches.
 - on Windows and Linux, the default CLI contribution is also shadowed, but it's partially working with an absolute workspace root path.

**Comment by per1234** [comment]
I checked on Linux and got the same result as on Windows: the sketch opens when I pass the absolute path via a command line argument, but it does not open when I pass the relative path.

Even with absolute path, it only works when I pass the path to the `.ino` file. It does not work when I pass the path to the sketch folder. Unfortunate, but this is also how Arduino IDE 1.x works so I think the users are accustomed to that quirk.

**Comment by kittaakos** [comment]
> It does not work when I pass the path to the sketch folder.

I will fix this too as part of #1563
